### PR TITLE
Attach a custom tracking event when a post is created via the ShareExtension

### PR DESCRIPTION
**Pull request by colinta**
I think the number of lines added and removed is approximately equal to how many months ago the story was created.

*Progress!*

### CODE REVIEW

**Comment by steam** [comment]
:metal:

File: Sources/Networking/ElloAPISharingHeaders.swift [hunk 1]
```diff
@@ -0,0 +1,7 @@
+////
+///  ElloAPISharingHeaders.swift
+//
+
+extension ElloAPI {
+    var sharingHeaders: [String: String]? { return nil }
```
Comment (colinta) [inline Sources/Networking/ElloAPISharingHeaders.swift]: No-op in the app, so all good there.

File: Sources/Networking/ElloAPI.swift [hunk 2]
```diff
@@ -472,6 +472,10 @@ extension ElloAPI: Moya.TargetType {
             ]
         }
 
+        if let sharingHeaders = self.sharingHeaders {
+            assigned += sharingHeaders
```
Comment (colinta) [inline Sources/Networking/ElloAPI.swift]: Found headers, add them to the request.

File: ShareExtension/ElloAPISharingHeaders.swift [hunk 3]
```diff
@@ -0,0 +1,14 @@
+////
+///  ElloAPISharingHeaders.swift
+//
+
+extension ElloAPI {
+    var sharingHeaders: [String: String]? {
+        switch self {
+        case .CreatePost:
+            return ["X-Segment-Event": "share-event"]
```
Comment (colinta) [inline ShareExtension/ElloAPISharingHeaders.swift]: The `X-Segment-Event` header is *only* added for the `CreatePost` endpoint.

**Pull request merged**